File: apps/android/app/src/main/kotlin/com/nogeass/passalarm/domain/usecase/DeletePlanUseCase.kt
package com.nogeass.passalarm.domain.usecase

import com.nogeass.passalarm.domain.repository.AlarmPlanRepository
import javax.inject.Inject

class DeletePlanUseCase @Inject constructor(
    private val planRepository: AlarmPlanRepository,
    private val reschedule: RescheduleNextNUseCase
) {
    suspend fun execute(planId: Long) {
        planRepository.delete(planId)
        reschedule()
    }
}

File: apps/ios/PassAlarm/Domain/UseCase/DeletePlanUseCase.swift
import Foundation

final class DeletePlanUseCase: Sendable {
    private let planRepository: AlarmPlanRepositoryProtocol
    private let reschedule: RescheduleNextNUseCase

    init(planRepository: AlarmPlanRepositoryProtocol,
         reschedule: RescheduleNextNUseCase) {
        self.planRepository = planRepository
        self.reschedule = reschedule
    }

    func execute(planId: UUID) async throws {
        try await planRepository.delete(planId)
        try await reschedule.execute()
    }
}
